server/src/controllers/course.ts

```diff
@@ -23,181 +25,142 @@ export async function addCourse(req: Request, res: Response): Promise<void> {
-      },
-      {
-        courseId: course.id,
-        language: Language.Swedish,
-        department: req.body.department.sv ?? '',
-        courseName: req.body.name.sv ?? ''
-      }
-    ], { transaction: t });
-
-    await t.commit();
-
-    const courseData: CourseData = {
-      id: course.id,
-      courseCode: course.courseCode,
-      department: {
-        en: '',
-        fi: '',
-        sv: ''
-      },
-      name: {
-        en: '',
-        fi: '',
-        sv: ''
-      },
-      evaluationInformation: {
-        en: '',
-        fi: '',
-        sv: ''
-      }
-    };
-
-    for (const translation of courseTranslations) {
-      switch (translation.language) {
-
-      case Language.English:
-        courseData.department.en = translation.department;
-        courseData.name.en = translation.courseName;
-        break;
-
-      case Language.Finnish:
-        courseData.department.fi = translation.department;
-        courseData.name.fi = translation.courseName;
-        break;
-
-      case Language.Swedish:
-        courseData.department.sv = translation.department;
-        courseData.name.sv = translation.courseName;
-        break;
-
-      }
+  const course: Course = await Course.create({
+    courseCode: req.body.courseCode
+  }, { transaction: t });
+
+  const courseTranslations: Array<CourseTranslation> = await CourseTranslation.bulkCreate([
+    {
+      courseId: course.id,
+      language: Language.Finnish,
+      department: req.body.department.fi ?? '',
+      courseName: req.body.name.fi ?? ''
+    },
+    {
+      courseId: course.id,
+      language: Language.English,
+      department: req.body.department.en ?? '',
+      courseName: req.body.name.en ?? ''
+    },
+    {
+      courseId: course.id,
+      language: Language.Swedish,
+      department: req.body.department.sv ?? '',
+      courseName: req.body.name.sv ?? ''
     }
+  ], { transaction: t });
+
+  await t.commit();
+
+  const courseData: CourseData = {
+    id: course.id,
+    courseCode: course.courseCode,
+    department: {
+      en: '',
+      fi: '',
+      sv: ''
+    },
+    name: {
+      en: '',
+      fi: '',
+      sv: ''
+    },
+    evaluationInformation: {
+      en: '',
+      fi: '',
+      sv: ''
+    }
+  };
+
+  for (const translation of courseTranslations) {
+    switch (translation.language) {
+
+    case Language.English:
+      courseData.department.en = translation.department;
+      courseData.name.en = translation.courseName;
+      break;
+
+    case Language.Finnish:
+      courseData.department.fi = translation.department;
+      courseData.name.fi = translation.courseName;
+      break;
+
+    case Language.Swedish:
+      courseData.department.sv = translation.department;
+      courseData.name.sv = translation.courseName;
+      break;
 
-    res.json({
-      success: true,
-      course: courseData
-    });
-  } catch (error) {
-    // TODO: appropriate logging in case of errors
-    await t.rollback();
```

Comment:
I think we've lost this rollback in the refactoring process. Sequelize does seem to provide a very handy way to automatically roll back, though: https://sequelize.org/docs/v6/other-topics/transactions/#managed-transactions